virttest/utils_misc.py

```diff
@@ -2105,19 +2105,27 @@ def get_cpu_vendor(cpu_info="", verbose=True):
     return vendor
 
 
-def get_support_machine_type(qemu_binary="/usr/libexec/qemu-kvm"):
+def get_support_machine_type(qemu_binary="/usr/libexec/qemu-kvm", remove_alias=False):
     """
-    Get the machine type the host support,return a list of machine type
+    Get the machine type the host support,return a tuple (s, c, v),
+    s, c, v are list, such as:
+    'pc', 'RHEL 7.4.0 PC (i440FX + PIIX, 1996)', '(alias of pc-i440fx-rhel7.4.0)'
     """
     o = process.system_output("%s -M ?" % qemu_binary).splitlines()
     s = []
     c = []
+    v = []
     for item in o[1:]:
-        machine_list = re.split(r"\s+", item, 1)
-        if machine_list[0] != "none":
-            s.append(machine_list[0])
-            c.append(machine_list[1])
-    return (s, c)
+        if remove_alias and "alias" in item:
+            continue
+        if "none" in item:
+            continue
+        split_pattern = re.compile(r'^(\S+)\s+(.*?)(?: (\((?:alias|default).*))?$')
```

Comment:
We can put this line out of `for` loop.

Reply:
@sitoliu  agree, updated, could you help review again ?  thanks.